sdks/python/apache_beam/yaml/yaml_udf_test.py

```diff
@@ -0,0 +1,325 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+import unittest
+from unittest import mock
+
+import apache_beam as beam
+from apache_beam.options import pipeline_options
+from apache_beam.testing.util import equal_to, assert_that
+from apache_beam.yaml.yaml_transform import YamlTransform
+
+MOCK_UDF_FILE_PREFIX = "/path/to/udf"
+
+
+class YamlUDFMappingTest(unittest.TestCase):
+  def __init__(self, method_name='runYamlMappingTest'):
+    super().__init__(method_name)
+    self.data = [
+        beam.Row(label='11a', conductor=11, rank=0),
+        beam.Row(label='37a', conductor=37, rank=1),
+        beam.Row(label='389a', conductor=389, rank=2),
+    ]
+    self.builtin_open = open  # save the unpatched version
+
+  class MockBlob:
+    def __init__(self, data):
+      self.data = data
+
+    def download_as_string(self):
+      return bytes(self.data, encoding='utf-8')
+
+  class MockBucket:
```

Comment:
If you use `Filesystems` you won't have to bother making all these mocks as it's already well-tested. (You could just do a single local file test with tempfile.)